### PR TITLE
[GCP] add compute datastream tests

### PR DESCRIPTION
**Pull request by endorama**
This PR is part of the effort to migrate Beats metricsets to data stream and adding system tests for them.

## What does this PR do?

- add GCP envs to Jenkins pipelines
- add gcp.compute datastream tests

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ]

## Related issues

- https://github.com/elastic/integrations/issues/490

### CODE REVIEW

**Comment by endorama** [comment]
CI failed with a permission error on the GCP Service Account:
>  Error: Error loading zone 'us-central1-a': googleapi: Error 403: Required 'compute.zones.get' permission for 'projects/elastic-obs-integrations-dev/zones/us-central1-a', forbidden

@v1v I'm using the same SA we setup in this PR https://github.com/elastic/elastic-package/pull/701 (see https://github.com/elastic/elastic-package/pull/701/files#diff-ef8b2f5e4d21d81c0eba4754dea3a3dfc2c67d26284f195165b650cf4aff7beaR23). Could it be the SA should be enabled for this repository?

**Comment by v1v** [comment]
> Could it be the SA should be enabled for this repository?

IIRC, @mtojek created the service account, though I'm inclined to suggest to use another service account with the right permissions instead, to avoid messing up the other pipelines/consumers.


I can provide you some details about how to update the secret in vault if needed

**Comment by endorama** [comment]
>  I'm inclined to suggest to use another service account with the right permissions instead, to avoid messing up the other pipelines/consumers.

I think we want this to be the same SA for this reason.

But there was a bug in the code that's why the permissions error showed up.

**Comment by mtojek** [comment]
> IIRC, @mtojek created the service account, though I'm inclined to suggest to use another service account with the right permissions instead, to avoid messing up the other pipelines/consumers.

I suggest using the same GCP service account as elastic-package uses for GCP testing. 

> elastic-obs-integrations-dev

Shouldn't it be `elastic-observability`?

@endorama Please try not using the force-push mode as you lose the commit history and we don't know what have you tried and what haven't.

**Comment by endorama** [comment]
@mtojek 

> I suggest using the same GCP service account as elastic-package uses for GCP testing (elastic-package-gcp).

That's the plan

> Shouldn't it be elastic-observability?

yes, there was a bug and the code didn't use the proper project, fixed in https://github.com/elastic/integrations/pull/2849/commits/9b6f981e423a2abfb1870a83aed2b4cdd18ebdd0

> @endorama Please try not using the force-push mode as you lose the commit history and we don't know what have you tried and what haven't.

You're right, sorry!

File: packages/gcp/manifest.yml [hunk 1]
```diff
@@ -50,64 +50,6 @@ vars:
     required: false
     show_user: true
 policy_templates:
-  - name: audit
```
[the remark this answers is not in the document]
Comment (endorama) [reply packages/gcp/manifest.yml:53]: No, but tests are failing with some `gcp-pubsub` [related](https://beats-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations/detail/PR-2849/7/pipeline#step-15862-log-70) failures and I wanted to remove that to observe if the issue was being introduced by these changeset.

Failure:
```
[2022-03-21T17:00:29.474Z] Error: error running package system tests: could not complete test run: could not add data stream config to policy: could not add package to policy; API status code = 500; response body = {"statusCode":500,"error":"Internal Server Error","message":"Input template not found, unable to find input type gcp-pubsub"}
```

**Comment by endorama** [comment]
Due to the data streams test errors I see on this PR that are unrelated to the changes in this PR I'm pausing the work and ensure the CI is green in `gcp-metrics` branch, the branch this PR is based on.

File: .ci/Jenkinsfile [hunk 2]
```diff
@@ -257,6 +259,11 @@ def withCloudTestEnv(Closure body) {
     [var: "AWS_ACCESS_KEY_ID", password: aws.access_key],
     [var: "AWS_SECRET_ACCESS_KEY", password: aws.secret_key],
   ])
+  // GCP
+  withGCPEnv(secret: env.ELASTIC_PACKAGE_GCP_SECRET) {
```
[the remark this answers is not in the document]
Comment (mtojek) [reply .ci/Jenkinsfile:263]: Here: https://github.com/elastic/apm-pipeline-library/blob/main/vars/withGCPEnv.groovy :)

**Pull request closed without merging**